server/src/test/java/org/elasticsearch/index/codec/CodecTests.java

```diff
@@ -96,6 +98,40 @@ public void testLegacyBestCompression() throws Exception {
         }
     }
 
+    public void testCodecRetrievalForUnknownCodec() throws Exception {
+        CodecService codecService = createCodecService();
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> codecService.codec("unknown_codec"));
+        assertEquals("failed to find codec [unknown_codec]", exception.getMessage());
+    }
+
+    public void testAvailableCodecsContainsExpectedCodecs() throws Exception {
+        CodecService codecService = createCodecService();
+        String[] availableCodecs = codecService.availableCodecs();
+        List<String> codecList = Arrays.asList(availableCodecs);
+
+        assertTrue(codecList.contains(CodecService.DEFAULT_CODEC));
+        assertTrue(codecList.contains(CodecService.LEGACY_DEFAULT_CODEC));
+        assertTrue(codecList.contains(CodecService.BEST_COMPRESSION_CODEC));
+        assertTrue(codecList.contains(CodecService.LEGACY_BEST_COMPRESSION_CODEC));
+        assertTrue(codecList.contains(CodecService.LUCENE_DEFAULT_CODEC));
+    }
+
+    public void testAvailableCodecsDoesNotContainUnknownCodec() throws Exception {
+        CodecService codecService = createCodecService();
+        String[] availableCodecs = codecService.availableCodecs();
+        List<String> codecList = Arrays.asList(availableCodecs);
+
+        assertFalse(codecList.contains("unknown_codec"));
+    }
+
+    public void testAvailableCodecsCount() throws Exception {
```

Comment:
This is very similar to `testAvailableCodecsContainsExpectedCodecs`, i don't think it adds much value. Same for the test above.

Reply:
I see that one solution could be to combine the checks into a single test to ensure that the expected codecs are present, that there are no unknown codecs, and to verify the total number of codecs. Should I update my PR to reflect this approach?

Reply:
I have made the requested changes and combined the checks into a single test. Is there anything else I can do to get the PR approved? Please let me know if there are any other modifications needed.
Thanks!